blockchain/difficulty.go

```diff
@@ -540,6 +540,34 @@ func (b *BlockChain) sumPurchasedTickets(startNode *blockNode, numToSum int64) i
 	return numPurchased
 }
 
+// LiveTicketsAtNextMaturity returns the total number of live tickets
+// available after the ticket maturity period from the provided block.
+// This assumes all blocks mined during the period are voted on
+// by the maximum number of tickets allowed per block.
+func (b *BlockChain) LiveTicketsAtNextMaturity(hash *chainhash.Hash) uint64 {
+	b.chainLock.Lock()
+	defer b.chainLock.Unlock()
+
+	node := b.index.lookupNode(hash)
+	liveTickets := int64(node.stakeNode.PoolSize())
```

Comment:
This is the only call that needs the lock.  The rest are accessing immutable fields.

Also this will crash if called with a hash that doesn't exist and it isn't checking if the data is available to properly support pruning.  I suggest taking a look at something like `BlockByHash` for an example.

e.g.

```Go
	node := b.index.LookupNode(hash)
	if node == nil || !b.index.NodeStatus(node).HaveData() {
		return nil, fmt.Errorf("block %s is not known", hash)
	}
	
	b.chainLock.Lock()
	liveTickets := int64(node.stakeNode.PoolSize())
	b.chainLock.Unlock()

	...
```